qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -122,20 +151,81 @@ def excitations(
-        return self._gsc.get_qubit_operators(problem, aux_operators)
+    ) -> tuple[PauliSumOp, Optional[dict[str, PauliSumOp]]]:
+        """Gets the operator and auxiliary operators, and transforms the provided auxiliary operators.
+
+        Note that contrary to the method :meth:`get_qubit_operators` from the
+        :class:`GroundStateEigensolver`, this returns three outputs: the hamiltonian, second quantization
+        default auxiliaries, and user defined auxiliaries.
+        Also note that this methods performs a specific treatment of the symmetries required by the qEOM
+        calculation.
+
+        Args:
+            problem: A class encoding a problem to be solved.
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Returns:
+            Tuple of the form (Qubit operator, Auxiliary operators).
+        """
+
+        main_operator, aux_second_q_operators = problem.second_q_ops()
+
+        # 1. Convert the main operator (hamiltonian) to PauliSumOp and apply two qubit reduction
+        num_particles = getattr(problem, "num_particles", None)
+        main_op = self.qubit_converter.convert_only(
+            main_operator,
+            num_particles=num_particles,
+        )
+        self.qubit_converter.force_match(num_particles=num_particles)
+
+        aux_ops = self.qubit_converter.convert_match(aux_second_q_operators)
+        cast(ListOrDictType[PauliSumOp], aux_ops)
+        if aux_operators is not None:
+            for name, op in aux_operators.items():
+                if isinstance(op, (SparseLabelOp)):
+                    converted_aux_op = self.qubit_converter.convert_match(op)
+                else:
+                    converted_aux_op = op
+                aux_ops[name] = converted_aux_op
+
+        # 2. Find the z2symmetries, set them in the qubit_converter, and apply the first step of the
+        # tapering.
+        _, z2symmetries = self.qubit_converter.find_taper_op(
+            main_op, problem.symmetry_sector_locator
+        )
+        self.qubit_converter.force_match(z2symmetries=z2symmetries)
+
+        untap_main_op = self.qubit_converter.convert_clifford(main_op)
+        untap_aux_ops = self.qubit_converter.convert_clifford(aux_ops)
+
+        # 3. If the eigensolver does not support auxiliary operators, reset them
+        if not self.solver.supports_aux_operators():
+            untap_aux_ops = None
+
+        # 4. If a MinimumEigensolverFactory was provided, then an additional call to get_solver() is
+        # required.
+        if isinstance(self.solver, MinimumEigensolverFactory):
+            self._gsc._solver = self.solver.get_solver(problem, self.qubit_converter)
+
+        return untap_main_op, untap_aux_ops
 
     def solve(
         self,
         problem: BaseProblem,
-        aux_operators: Optional[dict[str, SparseLabelOp]] = None,
+        aux_operators: Optional[dict[str, Union[SparseLabelOp, PauliSumOp]]] = None,
```

Comment:
Actually we need to update all explicit mentions of `PauliSumOp` to instead use `qiskit_nature.second_q.algorithms.ground_state_solvers.ground_state_solver.QubitOperator`. The reason being that is the union of `PauliSumOp` (which will be pending deprecated soon) and `SparsePauliOp` which we need to focus on here :+1: